lib/ansible/plugins/inventory/k8s.py

```diff
@@ -108,15 +108,245 @@
     context: 'awx/192-168-64-4:8443/developer'
 '''
 
+import json
+
+from ansible.errors import AnsibleError
+from ansible.module_utils.k8s.common import K8sAnsibleMixin, HAS_K8S_MODULE_HELPER
 from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
-from ansible.module_utils.k8s.inventory import K8sInventoryHelper
 
+try:
+    from openshift.dynamic.exceptions import DynamicApiError
+except ImportError:
+    pass
+
+
+def format_dynamic_api_exc(exc):
+    if exc.body:
+        if exc.headers and exc.headers.get('Content-Type') == 'application/json':
+            message = json.loads(exc.body).get('message')
+            if message:
+                return message
+        return exc.body
+    else:
+        return '%s Reason: %s' % (exc.status, exc.reason)
+
+
+class K8sInventoryException(Exception):
+    pass
 
-class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable, K8sInventoryHelper):
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable, K8sAnsibleMixin):
     NAME = 'k8s'
 
+    transport = 'kubectl'
```

Comment:
What is this for? kubectl is not used at all by ansible, afaik

Reply:
I didnt' add it, it comes from: https://github.com/ansible/ansible/blob/f73996e6ebc25a453caef3631558fca6e1af4a4e/lib/ansible/module_utils/k8s/inventory.py#L40

It is used there in order to define connection plugin (`kubectl` or `oc`) which will be used with the generated inventory:
https://github.com/ansible/ansible/blob/f73996e6ebc25a453caef3631558fca6e1af4a4e/lib/ansible/module_utils/k8s/inventory.py#L161-L167

Reply:
`kubectl` is a valid connection plugin, without looking into the code further I assume that's being used as the transport backend in that inventory plugin.

Reply:
https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/connection/kubectl.py

Reply:
Yeah, `ansible_connection` [selects](https://docs.ansible.com/ansible/latest/user_guide/intro_inventory.html#non-ssh-connection-types) the connection plugin :)